bodo/transforms/distributed_analysis.py

```diff
@@ -881,6 +913,11 @@ def _analyze_call(
         self, inst, lhs, rhs, func_var, args, kws, equiv_set, array_dists
     ):
         """analyze array distributions in function calls"""
+        from bodo.transforms.distributed_analysis_call_registry import (
+            DistributedAnalysisContext,
```

Comment:
Do you know what's the source of this circular import pattern? It would be great if we could refactor a common building block to its own file so we can avoid this.

Reply:
`distributed_analysis_call_registry` imports helpers from `distributed_analysis.py`. Doesn't seem worth refactoring right now but will do in the future.